### PR TITLE
Update ftw.mail to fix issues with message/rfc822 attachments.

### PR DESCRIPTION
**Pull request by njohner**
We have introduced a few hacks when extracting attachment data from mails in Gever, notably unwrapping `message/rfc822` attachments. It makes sense to have these fixes in `ftw.mail`, so they can be used for the attachment download view, which did currently not handle such attachments correctly.
I also noticed that, because we have overwritten the whole `AttachmentView`, all changes needed to be made twice, both in `ftw.mail` and in `opengever.core` (e.g. https://github.com/4teamwork/ftw.mail/pull/58/commits/13a1c4a3c8becad9e0b2165943b93fadd11487a4 and https://github.com/4teamwork/opengever.core/pull/6490). I decided to extract the necessary functionality in `ftw.mail` to separate methods, so that we could overwrite those specifically in `opengever.core`, to avoid overwriting the whole View.

With these changes (and the bump of `ftw.mail`) we fix several issues:
- Handle mail attached to mail with missing filenames.
- Fix download of mail attached to mail.

Definition of Done: https://4teamwork.atlassian.net/wiki/spaces/CHX4TW/pages/917562/

For [CA-2375]

## Checklist

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

[CA-2375]: https://4teamwork.atlassian.net/browse/CA-2375

### CODE REVIEW

File: changes/CA-2375.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Update ftw.mail to fix issues with message/rfc822 attachments.
```
Comment (deiferni) [inline changes/CA-2375.bugfix:1]: add your name 😄 ?

File: opengever/mail/browser/attachment.py [hunk 2]
```diff
@@ -1,71 +1,17 @@
...
     """Returns the attachment at the position specified in the request.
     """
 
-    def render(self):
-        """ Copied from ftw.mail.attachment:
-        Just add the contenttype-check (see l.51)
-        """
+    def set_content_type(self, content_type, filename):
+        if content_type == 'application/octet-stream':
+            mtr = getToolByName(self.context, 'mimetypes_registry')
```
Comment (deiferni) [inline opengever/mail/browser/attachment.py:12]: use `plone.api` to get tools?

File: sources.cfg [hunk 3]
```diff
@@ -8,6 +8,7 @@ extensions += mr.developer
 development-packages =
   opengever.maintenance
   collective.js.timeago
+  ftw.mail
```
Comment (deiferni) [inline sources.cfg:11]: ✂️

File: sources.cfg [hunk 4]
```diff
@@ -18,3 +19,4 @@ auto-checkout = ${buildout:development-packages}
 [branches]
 Products.LDAPUserFolder = ftw
 Products.LDAPMultiPlugins = ftw
+ftw.mail = nj/CA-2375/fix_attached_multipart_eml
```
Comment (deiferni) [inline sources.cfg:22]: ✂️

**Review by deiferni — APPROVED** [review]
⚠️ you need to remove ftw.mail form sources/auto checkout before merging!

Apart from that and some nagging LGTM 👍.

File: opengever/mail/browser/attachment.py (hunk 2 above)
Comment (njohner) [reply opengever/mail/browser/attachment.py:12]: Sure why not...

**Pull request merged**